tests/conftest.py

```diff
@@ -175,7 +178,25 @@ def format_setup_cfg(deps: Deps, no_extras: ExtraDeps) -> str:
             + "\n".join(f"    {d}" for d in deps)
         )
 
-    def format_pyproject_toml(deps: Deps, extras: ExtraDeps) -> str:
+    def format_pyproject_toml(deps: Deps, extras: ExtraDeps, is_dynamic: bool) -> str:
```

Comment:
I feel uneasy about re-using the `deps` and `extras` argument to pass the dynamic requirements files as, although these are technically `List[str]` and `Dict[str, Deps]` respectively, they are not dependencies but rather paths. Would it be possible to define new arguments to hold these requirements files?
Incidentally, that would also make it possible to define both static and dynamic dependencies in the `pyproject.toml` file (see my comment below).